### PR TITLE
Move SIP security and monitor features to OL

### LINKED ISSUE #15096: Move SIP security and monitor features to OL

### PR DESCRIPTION
**Pull request by annaguri**
fixes #15096

### CODE REVIEW

**Comment by annaguri** [comment]
#build

**Comment by annaguri** [comment]
#run-libby-bot

**Comment by annaguri** [comment]
#build

**Comment by annaguri** [comment]
#build

**Comment by annaguri** [comment]
#build

**Comment by annaguri** [comment]
#run-libby-bot

**Comment by tevans78** [comment]
@annaguri Do you have a plan to move the FAT buckets over as well? While it does not have to be done in the same PR, you should look into that.

**Comment by annaguri** [comment]
@tevans78 we are planning to move the FAT buckets as well, but it'll be done in a separate PR.

File: dev/com.ibm.ws.sipcontainer.monitor/.classpath [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.org.osgi.annotation.versioning"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.monitor"/>
+	<classpathentry kind="lib" path="/prereq.published/lib/com.ibm.ws.sipcontainer.jar"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.org.osgi.core"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.org.osgi.service.component"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.wsspi.org.osgi.service.component.annotations"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.logging"/>
+	<classpathentry kind="output" path="build/classes"/>
+</classpath>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.monitor/.classpath:13]: It should just be:
```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src"/>
	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
	<classpathentry kind="output" path="bin"/>
</classpath>
```

File: dev/com.ibm.ws.sipcontainer.security/.classpath [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	<classpathentry kind="src" path="src"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.security.authentication.tai"/>
+	<classpathentry kind="lib" path="/prereq.published/lib/com.ibm.ws.sipcontainer.jar"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.security"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.javaee.servlet.3.0"/>
+	<classpathentry kind="lib" path="/prereq.published/lib/com.ibm.websphere.javaee.servlet.sip.1.1.jar"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.org.osgi.annotation.versioning"/>
+	<classpathentry kind="output" path="build/classes"/>
+</classpath>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/.classpath:12]: It should just be:
```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src"/>
	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
	<classpathentry kind="output" path="bin"/>
</classpath>
```

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/.classpath [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.webcontainer"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.javaee.servlet.3.1"/>
+	<classpathentry kind="lib" path="/prereq.published/lib/com.ibm.websphere.javaee.servlet.sip.1.1.jar"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.container.service"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.org.osgi.core"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.websphere.org.osgi.service.component"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.wsspi.org.osgi.service.component.annotations"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/com.ibm.ws.session"/>
+	<classpathentry kind="output" path="build/classes"/>
+</classpath>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/.classpath:17]: It should just be:
```
<?xml version="1.0" encoding="UTF-8"?>
<classpath>
	<classpathentry kind="src" path="src"/>
	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
	<classpathentry kind="output" path="bin"/>
</classpath>
```

File: dev/com.ibm.ws.sipcontainer.monitor/src/com/ibm/ws/sip/monitor/pmi/application/ApplicationsModule.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+import com.ibm.ws.sip.container.pmi.ApplicationModuleInterface;
+import com.ibm.ws.sip.container.pmi.listener.ApplicationsPMIListener;
+
+@Component(configurationPolicy = ConfigurationPolicy.IGNORE)
+
+/**
+ * This class is a listener to SIP loaded applications. Each time the SIP Container loads a new SIP application
+ * it calls getApplicationModuel() method that creates a module for this application to manage the application counters.
+ * 
+ * @author galina
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.monitor/src/com/ibm/ws/sip/monitor/pmi/application/ApplicationsModule.java:34]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.monitor/src/com/ibm/ws/sip/monitor/pmi/basic/SipContainerBasicCountersMonitor.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+import com.ibm.websphere.monitor.annotation.PublishedMetric;
+import com.ibm.websphere.monitor.meters.Meter;
+import com.ibm.websphere.monitor.meters.MeterCollection;
+
+/**
+ * Monitor Class for the sip container basic counters.
+ * 
+ * This class is responsible for calculating and reporting SIP basic counters provides by the <code>SipContainerBasicCountersMXBean</code>.
+ * 
+ * @author galina, Sep 29, 2015
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.monitor/src/com/ibm/ws/sip/monitor/pmi/basic/SipContainerBasicCountersMonitor.java:26]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.security/src/com/ibm/websphere/security/tai/extension/HTTPTrustAssociationInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+ * <p> 
+ * The custom TAI must implement the following methods:
+ * <p>
+ * <code>initialize</code> to allocate any resources needed for the TAI
+ * <p>
+ * <code>negotiateValidateandEstablishProtocolTrust</code> that returns <code>TAIResult</code> to indicate the status of the HTTP message being processed
+ * <p> 
+ * <code>isTargetProtocolInterceptor</code> that returns false/true to indicate whether a HTTP message will be handled by the TAI. 
+ * 
+ * @author Michael Gilfix (<a href="mailto:mgilfix@us.ibm.com">mgilfix@us.ibm.com</a>)
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/src/com/ibm/websphere/security/tai/extension/HTTPTrustAssociationInterceptor.java:33]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.security/src/com/ibm/websphere/security/tai/extension/SIPTrustAssociationInterceptor.java [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+ * <p> 
+ * <code>isTargetProtocolInterceptor</code> that returns false/true to indicate whether a SIP message will be handled by the TAI. 
+ * <p>
+ * 
+ * The idea is during processing a SIP request, the Liberty server will pass the <code>
+ * SipServletRequest</code> and <code>SipServletResponse</code> to the SIP trust association interceptor. 
+ * The trust association interceptor can inspect the SIP message to see if it contains security
+ * attributes (authentication or authorization attributes) from the third party security service.
+ * </p>
+ * @author Michael Gilfix (<a href="mailto:mgilfix@us.ibm.com">mgilfix@us.ibm.com</a>)
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/src/com/ibm/websphere/security/tai/extension/SIPTrustAssociationInterceptor.java:42]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.security/src/com/ibm/wsspi/security/tai/extension/BaseTrustAssociationInterceptor.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+import com.ibm.websphere.security.tai.extension.SIPTrustAssociationInterceptor;
+import com.ibm.websphere.security.WebTrustAssociationFailedException;
+import com.ibm.ws.sip.container.was.ThreadLocalStorage;
+import com.ibm.wsspi.security.tai.TAIResult;
+import com.ibm.wsspi.security.tai.TrustAssociationInterceptor;
+
+/**
+ * This abstract <code>BaseTrustAssociationInterceptor</code> class provides a base implementation of the <code>TrustAssociationInterceptor</code> interface.
+ * 
+ * @author Michael Gilfix (<a href="mailto:mgilfix@us.ibm.com">mgilfix@us.ibm.com</a>)
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/src/com/ibm/wsspi/security/tai/extension/BaseTrustAssociationInterceptor.java:28]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/SipMessage31.java [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+package com.ibm.ws.sip.container.was.servlet31;
+
+import com.ibm.websphere.servlet31.request.IRequest31;
+import com.ibm.websphere.servlet31.response.IResponse31;
+import com.ibm.ws.sip.container.was.message.SipMessage;
+
+/**
+ * A version of SIP message that implements servlet31 request\response.
+ * Only used to add the inheritance without any actual content.
+ * @author sagia
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/SipMessage31.java:20]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/SipMessageFactory31.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+
+import com.ibm.ws.sip.container.was.message.SipMessage;
+import com.ibm.ws.sip.container.was.message.SipMessageFactory;
+
+/**
+ * Service that generate servlets31 sip messages.
+ * The service is used in the case that servlets31 feature is enabled in addition to the sip container.
+ * It will replace the SipMessageFactory and will use the servlets31 message.
+ * @author sagia
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/SipMessageFactory31.java:24]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/converged/SessionContextRegistryConverged31Impl.java [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+import com.ibm.ws.webcontainer.session.IHttpSessionContext;
+import com.ibm.wsspi.sip.converge.ConvergedHttpSessionContextImpl;
+import com.ibm.wsspi.sip.converge.SessionContextRegistryConvergedImpl;
+
+/**
+ * Extends the SessionContextRegistryConvergedImpl and providing ConvergedApplication (SIP and HTTP)
+ * ability.
+ * Support for Servlet 3.1 envirnoment.
+ * 
+ * @author anatf
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/src/com/ibm/ws/sip/container/was/servlet31/converged/SessionContextRegistryConverged31Impl.java:25]: Should remove reference to @author

File: dev/com.ibm.ws.sipcontainer.monitor/.project [hunk 12]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.monitor/.project:1]: Should have .settings/bndtools.core.prefs and .settings/org.eclipse.jdt.core.prefs files for this new component.

File: dev/com.ibm.ws.sipcontainer.security/.project [hunk 13]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/.project:1]: Should have .settings/bndtools.core.prefs and .settings/org.eclipse.jdt.core.prefs files for this new component.

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/.project [hunk 14]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/.project:1]: Should have .settings/bndtools.core.prefs and .settings/org.eclipse.jdt.core.prefs files for this new component.

File: dev/com.ibm.ws.sipcontainer.monitor/build.gradle [hunk 15]
```diff
@@ -0,0 +1,15 @@
+/*******************************************************************************
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.monitor/build.gradle:1]: This file is not needed in OL if it only has this content.  Please remove it.

File: dev/com.ibm.ws.sipcontainer.security/build.gradle [hunk 16]
```diff
@@ -0,0 +1,15 @@
+/*******************************************************************************
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.security/build.gradle:1]: This file is not needed in OL if it only has this content.  Please remove it.

File: dev/com.ibm.ws.sipcontainer.servlet.3.1/build.gradle [hunk 17]
```diff
@@ -0,0 +1,15 @@
+/*******************************************************************************
```
Comment (jhanders34) [inline dev/com.ibm.ws.sipcontainer.servlet.3.1/build.gradle:1]: This file is not needed in OL if it only has this content.  Please remove it.

**Review by jhanders34 — CHANGES_REQUESTED** [review]
I assumed the content of the classes were the same as they have been previously in the commercial WebSphere liberty source control and did not focus on the content of the classses.  If that is not the case and you want me to look at class changes, let me know which ones were changed.

**Comment by annaguri** [comment]
@jhanders34  the content of the classes have not changed. They are the same as in WL.

**Review by jhanders34 — APPROVED** [review]

**Comment by annaguri** [comment]
#build

**Review by jhanders34 — CHANGES_REQUESTED** [review]
oop

**Review by jhanders34 — APPROVED** [review]

File: dev/com.ibm.websphere.appserver.api.sipServletSecurity.1.0/bnd.bnd [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion: 1.0
+
+Bundle-Name: WebSphere SIP Container security API
+Bundle-Description: WebSphere SIP Container security Support, version ${bVersion}
+Bundle-SymbolicName: com.ibm.websphere.appserver.api.sipServletSecurity.1.0
+
+Import-Package: com.ibm.websphere.security.tai.extension
+
+Export-Package: com.ibm.websphere.security.tai.extension
```
Comment (jhanders34) [inline dev/com.ibm.websphere.appserver.api.sipServletSecurity.1.0/bnd.bnd:20]: The com.ibm.wsspi.security.tai.extension also needs to be exported.

**Review by jhanders34 — CHANGES_REQUESTED** [review]

File: dev/com.ibm.websphere.appserver.api.sipServletSecurity.1.0/bnd.bnd (hunk 18 above)
Comment (annaguri) [reply dev/com.ibm.websphere.appserver.api.sipServletSecurity.1.0/bnd.bnd]: done

**Review by jhanders34 — APPROVED** [review]

**Comment by jhanders34** [comment]
#build

**Comment by cbridgha** [comment]
#libby

**Pull request merged**